EventAwareLumiBased: unit test broken

Hi,

I have just noticed that this commit f93ba26014420fe96bbd127beb225ab415f0163f broke the unit test EventAwareLumiBasedTest.testD_NoFileSplitNoHardLimit. I am gatting this when executing it:

```
[mmascher@mmascher-dev6 WMCore]$ python test/python/WMCore_t/JobSplitting_t/EventAwareLumiBased_t.py EventAwareLumiBasedTest.testD_NoFileSplitNoHardLimitF
======================================================================
FAIL: testD_NoFileSplitNoHardLimit (__main__.EventAwareLumiBasedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/python/WMCore_t/JobSplitting_t/EventAwareLumiBased_t.py", line 346, in testD_NoFileSplitNoHardLimit
    self.assertEqual(len(jobs), 8, "Eight jobs must be in the jobgroup")
AssertionError: Eight jobs must be in the jobgroup

----------------------------------------------------------------------
Ran 1 test in 0.115s

FAILED (failures=1)
```

I haven't tried to understand what is wrong, I just wanted to notify this. @amaltaro , can you have a look?

--- a/test/python/WMCore_t/JobSplitting_t/EventAwareLumiBased_t.py
+++ b/test/python/WMCore_t/JobSplitting_t/EventAwareLumiBased_t.py
@@ -9,7 +9,6 @@ Created on Sep 25, 2012
 
 @author: dballest
 """
-
 import unittest
 
 from WMCore.DataStructs.File import File
@@ -21,13 +20,13 @@ from WMCore.DataStructs.Run import Run
 from WMCore.JobSplitting.SplitterFactory import SplitterFactory
 from WMCore.Services.UUIDLib import makeUUID
 
+
 class EventAwareLumiBasedTest(unittest.TestCase):
     """
     _EventAwareLumiBasedTest_
 
     Test event based job splitting.
     """
-
 
     def setUp(self):
         """
@@ -38,9 +37,9 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
 
         self.testWorkflow = Workflow()
-        self.performanceParams = {'timePerEvent' : 12,
-                                  'memoryRequirement' : 2300,
-                                  'sizePerEvent' : 400}
+        self.performanceParams = {'timePerEvent': 12,
+                                  'memoryRequirement': 2300,
+                                  'sizePerEvent': 400}
 
         return
 
@@ -52,8 +51,7 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
         pass
 
-
-    def createSubscription(self, nFiles, lumisPerFile, twoSites = False, nEventsPerFile = 100):
+    def createSubscription(self, nFiles, lumisPerFile, twoSites=False, nEventsPerFile=100):
         """
         _createSubscription_
 
@@ -62,7 +60,7 @@ class EventAwareLumiBasedTest(unittest.TestCase):
 
         baseName = makeUUID()
 
-        testFileset = Fileset(name = baseName)
+        testFileset = Fileset(name=baseName)
         for i in range(nFiles):
             newFile = self.createFile('%s_%i' % (baseName, i), nEventsPerFile,
                                       i, lumisPerFile, 'blenheim')
@@ -73,11 +71,10 @@ class EventAwareLumiBasedTest(unittest.TestCase):
                                           i, lumisPerFile, 'malpaquet')
                 testFileset.addFile(newFile)
 
-
-        testSubscription = Subscription(fileset = testFileset,
-                                         workflow = self.testWorkflow,
-                                         split_algo = "EventAwareLumiBased",
-                                         type = "Processing")
+        testSubscription = Subscription(fileset=testFileset,
+                                        workflow=self.testWorkflow,
+                                        split_algo="EventAwareLumiBased",
+                                        type="Processing")
 
         return testSubscription
 
@@ -87,8 +84,8 @@ class EventAwareLumiBasedTest(unittest.TestCase):
 
         Create a file for testing
         """
-        newFile = File(lfn = lfn, size = 1000,
-                       events = events)
+        newFile = File(lfn=lfn, size=1000,
+                       events=events)
         lumiList = []
         for lumi in range(lumis):
             lumiList.append((run * lumis) + lumi)
@@ -104,41 +101,35 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
         splitter = SplitterFactory()
 
-        oneSetSubscription = self.createSubscription(nFiles = 10, lumisPerFile = 1)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = oneSetSubscription)
+        oneSetSubscription = self.createSubscription(nFiles=10, lumisPerFile=1)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=oneSetSubscription)
 
-
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               events_per_job = 100,
-                               performance = self.performanceParams)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               events_per_job=100,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1)
         self.assertEqual(len(jobGroups[0].jobs), 10)
         for job in jobGroups[0].jobs:
             self.assertTrue(len(job['input_files']), 1)
 
-
-
-
-        twoLumiFiles = self.createSubscription(nFiles = 5, lumisPerFile = 2)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = twoLumiFiles)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               events_per_job = 50,
-                               performance = self.performanceParams)
+        twoLumiFiles = self.createSubscription(nFiles=5, lumisPerFile=2)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=twoLumiFiles)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               events_per_job=50,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1)
         self.assertEqual(len(jobGroups[0].jobs), 10)
         for job in jobGroups[0].jobs:
             self.assertEqual(len(job['input_files']), 1)
 
-
-
-        wholeLumiFiles = self.createSubscription(nFiles = 5, lumisPerFile = 3)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = wholeLumiFiles)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               events_per_job = 67,
-                               performance = self.performanceParams)
+        wholeLumiFiles = self.createSubscription(nFiles=5, lumisPerFile=3)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=wholeLumiFiles)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               events_per_job=67,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1)
         # 10 because we split on run boundaries
         self.assertEqual(len(jobGroups[0].jobs), 10)
@@ -146,7 +137,6 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         for job in jobList:
             # Have should have one file, half two
             self.assertTrue(len(job['input_files']) in [1, 2])
-
 
         mask0 = jobList[0]['mask'].getRunAndLumis()
         self.assertEqual(mask0, {0: [[0, 1]]})
@@ -160,18 +150,16 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         self.assertEqual(jobList[0]['mask'].getRunAndLumis(), {0: [[0, 1]]})
 
         # Do it with multiple sites
-        twoSiteSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 2, twoSites = True)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = twoSiteSubscription)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               events_per_job = 50,
-                               performance = self.performanceParams)
+        twoSiteSubscription = self.createSubscription(nFiles=5, lumisPerFile=2, twoSites=True)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=twoSiteSubscription)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               events_per_job=50,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 2)
         self.assertEqual(len(jobGroups[0].jobs), 10)
         for job in jobGroups[0].jobs:
             self.assertEqual(len(job['input_files']), 1)
-
-
 
     def testB_NoRunNoFileSplitting(self):
         """
@@ -181,14 +169,14 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         cases that might be required.
         """
         splitter = SplitterFactory()
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 60,
-                               performance = self.performanceParams)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=60,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1)
         jobs = jobGroups[0].jobs
@@ -202,13 +190,13 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         self.assertEqual(job1runLumi[1][0][0], job1runLumi[1][0][1])  # Run 1, startLumi == endLumi
 
         # Assert that this works differently with file splitting on and run splitting on
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 60,
-                               performance = self.performanceParams)
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=60,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1)
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 10)
@@ -220,49 +208,30 @@ class EventAwareLumiBasedTest(unittest.TestCase):
 
         # Test total_events limit. (The algorithm cuts off after the lumi that
         # brings the total average event count over -or equal to- total_events.)
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 60,
-                               total_events = 10,
-                               performance = self.performanceParams)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=60,
+                               total_events=10,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1)
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 1)
         self.assertEqual(jobs[0]['mask'].getRunAndLumis(), {0: [[0, 0]]})
 
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 60,
-                               total_events = 179,
-                               performance = self.performanceParams)
-
-        self.assertEqual(len(jobGroups), 1)
-        jobs = jobGroups[0].jobs
-        self.assertEqual(len(jobs), 3)
-        self.assertEqual(jobs[0]['mask'].getRunAndLumis(), {0: [[0, 2]]})
-        job1runLumi = jobs[1]['mask'].getRunAndLumis()
-        self.assertEqual(job1runLumi[0][0][0] + 1, job1runLumi[0][0][1])  # Run 0, startLumi+1 == endLumi
-        self.assertEqual(job1runLumi[1][0][0], job1runLumi[1][0][1])  # Run 1, startLumi == endLumi
-        self.assertEqual(len(jobs[2]['mask'].getRunAndLumis()), 1)  # All lumis from one run again
-
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 60,
-                               total_events = 180,
-                               performance = self.performanceParams)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=60,
+                               total_events=179,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1)
         jobs = jobGroups[0].jobs
@@ -273,15 +242,34 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         self.assertEqual(job1runLumi[1][0][0], job1runLumi[1][0][1])  # Run 1, startLumi == endLumi
         self.assertEqual(len(jobs[2]['mask'].getRunAndLumis()), 1)  # All lumis from one run again
 
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 60,
-                               total_events = 181,
-                               performance = self.performanceParams)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=60,
+                               total_events=180,
+                               performance=self.performanceParams)
+
+        self.assertEqual(len(jobGroups), 1)
+        jobs = jobGroups[0].jobs
+        self.assertEqual(len(jobs), 3)
+        self.assertEqual(jobs[0]['mask'].getRunAndLumis(), {0: [[0, 2]]})
+        job1runLumi = jobs[1]['mask'].getRunAndLumis()
+        self.assertEqual(job1runLumi[0][0][0] + 1, job1runLumi[0][0][1])  # Run 0, startLumi+1 == endLumi
+        self.assertEqual(job1runLumi[1][0][0], job1runLumi[1][0][1])  # Run 1, startLumi == endLumi
+        self.assertEqual(len(jobs[2]['mask'].getRunAndLumis()), 1)  # All lumis from one run again
+
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=60,
+                               total_events=181,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1)
         jobs = jobGroups[0].jobs
@@ -305,69 +293,69 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
         splitter = SplitterFactory()
 
-        #Create 5 files with 7 lumi per file and 100 events per lumi on average.
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 7, twoSites = False,
-                                                   nEventsPerFile = 700)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        # Create 5 files with 7 lumi per file and 100 events per lumi on average.
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=7, twoSites=False,
+                                                   nEventsPerFile=700)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        #First test, the optimal settings are 360 events per job
-        #As we have files with 100 events per lumi, this will configure the splitting to
-        #3.6 lumis per job, which rounds to 3, the algorithm always approximates to the lower integer.
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 360,
-                               performance = self.performanceParams)
+        # First test, the optimal settings are 360 events per job
+        # As we have files with 100 events per lumi, this will configure the splitting to
+        # 3.6 lumis per job, which rounds to 3, the algorithm always approximates to the lower integer.
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=360,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 15, "There should be 15 jobs")
 
-        #Now set the average to 200 events per job
-        #This results in the algorithm reducing the lumis per job to 2
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 200,
-                               performance = self.performanceParams)
+        # Now set the average to 200 events per job
+        # This results in the algorithm reducing the lumis per job to 2
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=200,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 20, "There should be 20 jobs")
 
-        #Check extremes, process a zero event files with lumis. It must be processed in one job
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 100, twoSites = False,
-                                                   nEventsPerFile = 0)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               events_per_job = 5000,
-                               performance = self.performanceParams)
+        # Check extremes, process a zero event files with lumis. It must be processed in one job
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=100, twoSites=False,
+                                                   nEventsPerFile=0)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               events_per_job=5000,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 5, "There should be 5 jobs")
 
-        #Process files with 10k events per lumi, fallback to one lumi per job. We can't do better
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 5, twoSites = False,
-                                                   nEventsPerFile = 50000)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 5000,
-                               performance = self.performanceParams)
+        # Process files with 10k events per lumi, fallback to one lumi per job. We can't do better
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=5, twoSites=False,
+                                                   nEventsPerFile=50000)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=5000,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 25, "There should be 5 jobs")
 
         # Test total_events limit. (The algorithm cuts off after the lumi that
         # brings the total average event count over -or equal to- total_events.)
-        testSubscription = self.createSubscription(nFiles = 5, lumisPerFile = 3, twoSites = False,
-                                                   nEventsPerFile = 300)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 250,
-                               total_events = 750,
-                               performance = self.performanceParams)
+        testSubscription = self.createSubscription(nFiles=5, lumisPerFile=3, twoSites=False,
+                                                   nEventsPerFile=300)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=250,
+                               total_events=750,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 5, "There should be 5 jobs")
@@ -389,33 +377,33 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
         splitter = SplitterFactory()
 
-        #Create 100 files with 7 lumi per file and 0 events per lumi on average.
-        testSubscription = self.createSubscription(nFiles = 100, lumisPerFile = 7, twoSites = False,
-                                                   nEventsPerFile = 0)
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        # Create 100 files with 7 lumi per file and 0 events per lumi on average.
+        testSubscription = self.createSubscription(nFiles=100, lumisPerFile=7, twoSites=False,
+                                                   nEventsPerFile=0)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        #First test, the optimal settings are 360 events per job
-        #As we have files with 0 events per lumi, this will configure the splitting to
-        #a single job containing all files
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 360,
-                               performance = self.performanceParams)
+        # First test, the optimal settings are 360 events per job
+        # As we have files with 0 events per lumi, this will configure the splitting to
+        # a single job containing all files
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=360,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 1, "There should be 1 job")
         self.assertEqual(len(jobs[0]['input_files']), 100, "All 100 files must be in the job")
 
-        #Create 7 files, each one with different lumi/event distributions
-        testFileset = Fileset(name = "FilesetA")
-        testFileA = self.createFile("/this/is/file1", 250, 0, 5, "blenheim")
-        testFileB = self.createFile("/this/is/file2", 600, 1, 1, "blenheim")
-        testFileC = self.createFile("/this/is/file3", 1200, 2, 2, "blenheim")
-        testFileD = self.createFile("/this/is/file4", 100, 3, 1, "blenheim")
-        testFileE = self.createFile("/this/is/file5", 30, 4, 1, "blenheim")
-        testFileF = self.createFile("/this/is/file6", 10, 5, 1, "blenheim")
-        testFileG = self.createFile("/this/is/file7", 151, 6, 3, "blenheim")
+        # Create 7 files, each one with different lumi/event distributions
+        testFileset = Fileset(name="FilesetA")
+        testFileA = self.createFile("/this/is/file1", 250, 0, 5, "blenheim")  # job1, job2
+        testFileB = self.createFile("/this/is/file2", 600, 1, 1, "blenheim")  # job3
+        testFileC = self.createFile("/this/is/file3", 1200, 2, 2, "blenheim")  # job4, job5
+        testFileD = self.createFile("/this/is/file4", 100, 3, 1, "blenheim")  # job6
+        testFileE = self.createFile("/this/is/file5", 30, 4, 1, "blenheim")  # job6
+        testFileF = self.createFile("/this/is/file6", 10, 5, 1, "blenheim")  # job6
+        testFileG = self.createFile("/this/is/file7", 151, 6, 3, "blenheim")  # job7
         testFileset.addFile(testFileA)
         testFileset.addFile(testFileB)
         testFileset.addFile(testFileC)
@@ -424,38 +412,37 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         testFileset.addFile(testFileF)
         testFileset.addFile(testFileG)
 
-        testSubscription = Subscription(fileset = testFileset,
-                                        workflow = self.testWorkflow,
-                                        split_algo = "EventAwareLumiBased",
-                                        type = "Processing")
+        testSubscription = Subscription(fileset=testFileset,
+                                        workflow=self.testWorkflow,
+                                        split_algo="EventAwareLumiBased",
+                                        type="Processing")
 
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        #Optimal settings are: jobs with 150 events per job
-        #This means, the first file must be splitted in 3 lumis per job which would leave room
-        #for another lumi in the second job, but the second file has a lumi too big for that
-        #The 3rd job only contains the second file, the fourth and fifth job split the third file
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 150,
-                               performance = self.performanceParams)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        # Optimal settings are: jobs with 150 events per job
+        # This means, the first file must be splitted in 3 lumis per job which would leave room
+        # for another lumi in the second job, but the second file has a lumi too big for that
+        # The 3rd job only contains the second file, the fourth and fifth job split the third file
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=150,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
-        self.assertEqual(len(jobs), 8, "Eight jobs must be in the jobgroup")
-        self.assertEqual(jobs[0]["mask"].getRunAndLumis(), {0 : [[0, 2]]}, "Wrong mask for the first job")
-        self.assertEqual(jobs[1]["mask"].getRunAndLumis(), {0 : [[3, 4]]}, "Wrong mask for the second job")
-        self.assertEqual(jobs[2]["mask"].getRunAndLumis(), {1 : [[1, 1]]}, "Wrong mask for the third job")
-        self.assertEqual(jobs[3]["mask"].getRunAndLumis(), {2 : [[4, 4]]}, "Wrong mask for the fourth job")
-        self.assertEqual(jobs[4]["mask"].getRunAndLumis(), {2 : [[5, 5]]}, "Wrong mask for the fifth job")
+        self.assertEqual(len(jobs), 7, "Seven jobs must be in the jobgroup")
+        self.assertEqual(jobs[0]["mask"].getRunAndLumis(), {0: [[0, 2]]}, "Wrong mask for the first job")
+        self.assertEqual(jobs[1]["mask"].getRunAndLumis(), {0: [[3, 4]]}, "Wrong mask for the second job")
+        self.assertEqual(jobs[2]["mask"].getRunAndLumis(), {1: [[1, 1]]}, "Wrong mask for the third job")
+        self.assertEqual(jobs[3]["mask"].getRunAndLumis(), {2: [[4, 4]]}, "Wrong mask for the fourth job")
+        self.assertEqual(jobs[4]["mask"].getRunAndLumis(), {2: [[5, 5]]}, "Wrong mask for the fifth job")
         self.assertEqual(jobs[5]["mask"].getRunAndLumis(),
-                         {3 : [[3, 3]], 4 : [[4, 4]], 5 : [[5, 5]]}, "Wrong mask for the sixth job")
-        self.assertEqual(jobs[6]["mask"].getRunAndLumis(), {6 : [[18, 19]]}, "Wrong mask for the seventh job")
-        self.assertEqual(jobs[7]["mask"].getRunAndLumis(), {6 : [[20, 20]]}, "Wrong mask for the seventh job")
-        #Test interactions of this algorithm with splitOnRun = True
-        #Make 2 files, one with 3 runs and a second one with the last run of the first
-        fileA = File(lfn = "/this/is/file1", size = 1000,
-                       events = 2400)
+                         {3: [[3, 3]], 4: [[4, 4]], 5: [[5, 5]]}, "Wrong mask for the sixth job")
+        self.assertEqual(jobs[6]["mask"].getRunAndLumis(), {6: [[18, 20]]}, "Wrong mask for the seventh job")
+        # Test interactions of this algorithm with splitOnRun = True
+        # Make 2 files, one with 3 runs and a second one with the last run of the first
+        fileA = File(lfn="/this/is/file1", size=1000,
+                     events=2400)
         lumiListA = []
         lumiListB = []
         lumiListC = []
@@ -470,21 +457,21 @@ class EventAwareLumiBasedTest(unittest.TestCase):
 
         fileB = self.createFile('/this/is/file2', 200, 3, 5, "malpaquet")
 
-        testFileset = Fileset(name = 'FilesetB')
+        testFileset = Fileset(name='FilesetB')
         testFileset.addFile(fileA)
         testFileset.addFile(fileB)
-        testSubscription = Subscription(fileset = testFileset,
-                                        workflow = self.testWorkflow,
-                                        split_algo = "EventAwareLumiBased",
-                                        type = "Processing")
+        testSubscription = Subscription(fileset=testFileset,
+                                        workflow=self.testWorkflow,
+                                        split_algo="EventAwareLumiBased",
+                                        type="Processing")
 
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        #The settings for this splitting are 700 events per job
-        jobGroups = jobFactory(splitOnRun = True,
-                               halt_job_on_file_boundaries = False,
-                               events_per_job = 700,
-                               performance = self.performanceParams)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        # The settings for this splitting are 700 events per job
+        jobGroups = jobFactory(splitOnRun=True,
+                               halt_job_on_file_boundaries=False,
+                               events_per_job=700,
+                               performance=self.performanceParams)
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 6, "Six jobs must be in the jobgroup")
@@ -499,8 +486,8 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
         splitter = SplitterFactory()
 
-        #Create 3 files, the one in the middle is a "bad" file
-        testFileset = Fileset(name = "FilesetA")
+        # Create 3 files, the one in the middle is a "bad" file
+        testFileset = Fileset(name="FilesetA")
         testFileA = self.createFile("/this/is/file1", 1000, 0, 5, "blenheim")
         testFileB = self.createFile("/this/is/file2", 1000, 1, 1, "blenheim")
         testFileC = self.createFile("/this/is/file3", 1000, 2, 2, "blenheim")
@@ -508,26 +495,26 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         testFileset.addFile(testFileB)
         testFileset.addFile(testFileC)
 
-        testSubscription = Subscription(fileset = testFileset,
-                                        workflow = self.testWorkflow,
-                                        split_algo = "EventAwareLumiBased",
-                                        type = "Processing")
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        #Settings are to split on job boundaries, to fail sing lumis with more than 800 events
-        #and to put 550 events per job
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 550,
-                               max_events_per_lumi = 800,
-                               performance = self.performanceParams)
+        testSubscription = Subscription(fileset=testFileset,
+                                        workflow=self.testWorkflow,
+                                        split_algo="EventAwareLumiBased",
+                                        type="Processing")
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        # Settings are to split on job boundaries, to fail sing lumis with more than 800 events
+        # and to put 550 events per job
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=550,
+                               max_events_per_lumi=800,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 6, "Six jobs must be in the jobgroup")
         self.assertTrue(jobs[3]['failedOnCreation'], "The job processing the second file should me marked for failure")
         self.assertEqual(jobs[3]['failedReason'], "File /this/is/file2 has too many events (1000) in 1 lumi(s)",
-                          "The reason for the failure is not accurate")
+                         "The reason for the failure is not accurate")
 
         return
 
@@ -540,8 +527,8 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         """
         splitter = SplitterFactory()
 
-        #Create 3 single-big-lumi files
-        testFileset = Fileset(name = "FilesetA")
+        # Create 3 single-big-lumi files
+        testFileset = Fileset(name="FilesetA")
         testFileA = self.createFile("/this/is/file1", 1000, 0, 1, "somese.cern.ch")
         testFileB = self.createFile("/this/is/file2", 1000, 1, 1, "somese.cern.ch")
         testFileC = self.createFile("/this/is/file3", 1000, 2, 1, "somese.cern.ch")
@@ -549,27 +536,29 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         testFileset.addFile(testFileB)
         testFileset.addFile(testFileC)
 
-        testSubscription = Subscription(fileset = testFileset,
-                                        workflow = self.testWorkflow,
-                                        split_algo = "EventAwareLumiBased",
-                                        type = "Processing")
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        #Settings are to split on job boundaries, to fail sing lumis with more than 800 events
-        #and to put 550 events per job
-        jobGroups = jobFactory(halt_job_on_file_boundaries = True,
-                               splitOnRun = True,
-                               events_per_job = 550,
-                               max_events_per_lumi = 800,
-                               performance = self.performanceParams)
+        testSubscription = Subscription(fileset=testFileset,
+                                        workflow=self.testWorkflow,
+                                        split_algo="EventAwareLumiBased",
+                                        type="Processing")
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        # Settings are to split on job boundaries, to fail sing lumis with more than 800 events
+        # and to put 550 events per job
+        jobGroups = jobFactory(halt_job_on_file_boundaries=True,
+                               splitOnRun=True,
+                               events_per_job=550,
+                               max_events_per_lumi=800,
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 3, "Three jobs must be in the jobgroup")
-        for i in range(1,4):
-            self.assertTrue(jobs[i - 1]['failedOnCreation'], "The job processing the second file should me marked for failure")
-            self.assertEqual(jobs[i - 1]['failedReason'], "File /this/is/file%d has too many events (1000) in 1 lumi(s)" % i,
-                          "The reason for the failure is not accurate")
+        for i in range(1, 4):
+            self.assertTrue(jobs[i - 1]['failedOnCreation'],
+                            "The job processing the second file should me marked for failure")
+            self.assertEqual(jobs[i - 1]['failedReason'],
+                             "File /this/is/file%d has too many events (1000) in 1 lumi(s)" % i,
+                             "The reason for the failure is not accurate")
 
         return
 
@@ -585,9 +574,9 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         # - file1 with 1 run  of 8 lumis
         # - file2 with 2 runs of 2 lumis each
         # - file3 with 1 run  of 5 lumis
-        fileA = File(lfn = "/this/is/file1", size = 1000, events = 800)
-        fileB = File(lfn = "/this/is/file2", size = 1000, events = 400)
-        fileC = File(lfn = "/this/is/file3", size = 1000, events = 500)
+        fileA = File(lfn="/this/is/file1", size=1000, events=800)
+        fileB = File(lfn="/this/is/file2", size=1000, events=400)
+        fileC = File(lfn="/this/is/file3", size=1000, events=500)
 
         lumiListA = []
         for lumi in range(8):
@@ -608,25 +597,25 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         fileC.addRun(Run(4, *lumiListC))
         fileC.setLocation("somese.cern.ch")
 
-        testFileset = Fileset(name = 'Fileset')
+        testFileset = Fileset(name='Fileset')
         testFileset.addFile(fileA)
         testFileset.addFile(fileB)
         testFileset.addFile(fileC)
 
-        testSubscription = Subscription(fileset = testFileset,
-                                        workflow = self.testWorkflow,
-                                        split_algo = "EventAwareLumiBased",
-                                        type = "Processing")
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        testSubscription = Subscription(fileset=testFileset,
+                                        workflow=self.testWorkflow,
+                                        split_algo="EventAwareLumiBased",
+                                        type="Processing")
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
         # Use a lumi-mask = {1: [[10,14]], 2: [[20,21]], 4: [[40,41]]}
-        jobGroups = jobFactory(halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               events_per_job = 850,
-                               runs = ['1', '2', '4'],
-                               lumis = ['10,14', '20,21', '40,41'],
-                               performance = self.performanceParams)
+        jobGroups = jobFactory(halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               events_per_job=850,
+                               runs=['1', '2', '4'],
+                               lumis=['10,14', '20,21', '40,41'],
+                               performance=self.performanceParams)
 
         self.assertEqual(len(jobGroups), 1, "There should be only one job group")
         jobs = jobGroups[0].jobs
@@ -642,21 +631,21 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         cross multiple files.
         """
         splitter = SplitterFactory()
-        testSubscription = self.createSubscription(nFiles = 2, lumisPerFile = 2,
-                                                   twoSites = False, nEventsPerFile = 150)
+        testSubscription = self.createSubscription(nFiles=2, lumisPerFile=2,
+                                                   twoSites=False, nEventsPerFile=150)
         files = testSubscription.getFileset().getFiles()
         self.assertEqual(len(files), 2)
         # at the moment we have two files with two lumis each:
         #  file0 has run0 and lumis 0,1. 150 events
         #  file1 has run1 and lumis 2,3. 150 evens
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
 
-        jobGroups = jobFactory(events_per_job = 50,
-                               halt_job_on_file_boundaries = False,
-                               splitOnRun = False,
-                               performance = self.performanceParams,
-                               applyLumiCorrection = False
+        jobGroups = jobFactory(events_per_job=50,
+                               halt_job_on_file_boundaries=False,
+                               splitOnRun=False,
+                               performance=self.performanceParams,
+                               applyLumiCorrection=False
                               )
 
         # The splitting algorithm will assume 75 events per lumi.
@@ -665,8 +654,8 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 4)
 
-        testSubscription = self.createSubscription(nFiles = 2, lumisPerFile = 2,
-                                           twoSites = False, nEventsPerFile = 150)
+        testSubscription = self.createSubscription(nFiles=2, lumisPerFile=2,
+                                                   twoSites=False, nEventsPerFile=150)
         files = testSubscription.getFileset().getFiles()
         # Now modifyng and adding duplicated lumis.
         for runObj in files[0]['runs']:
@@ -679,12 +668,12 @@ class EventAwareLumiBasedTest(unittest.TestCase):
             runObj.run = 0
             runObj.appendLumi(42)
         files[1]['locations'] = set(['blenheim'])
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        jobGroups = jobFactory(events_per_job = 50,
-                               halt_job_on_file_boundaries = True,
-                               performance = self.performanceParams,
-                               applyLumiCorrection = True)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        jobGroups = jobFactory(events_per_job=50,
+                               halt_job_on_file_boundaries=True,
+                               performance=self.performanceParams,
+                               applyLumiCorrection=True)
 
         # Now we will have:
         #   file0: Run0 and lumis [0, 1, 42]
@@ -706,10 +695,9 @@ class EventAwareLumiBasedTest(unittest.TestCase):
         self.assertEqual(jobs[3]['mask'].getRunAndLumis(), {0: [[2, 2]]})
         self.assertEqual(jobs[4]['mask'].getRunAndLumis(), {0: [[3, 3]]})
 
-
-        #Check that if the last two jobs have the same duplicated lumi you do not get an error
-        testSubscription = self.createSubscription(nFiles = 2, lumisPerFile = 2,
-                                           twoSites = False, nEventsPerFile = 150)
+        # Check that if the last two jobs have the same duplicated lumi you do not get an error
+        testSubscription = self.createSubscription(nFiles=2, lumisPerFile=2,
+                                                   twoSites=False, nEventsPerFile=150)
         files = testSubscription.getFileset().getFiles()
         # Now modifying and adding the same duplicated lumis in the Nth and Nth-1 jobs
         for runObj in files[0]['runs']:
@@ -720,17 +708,16 @@ class EventAwareLumiBasedTest(unittest.TestCase):
             runObj.run = 0
             runObj.lumis = [42]
         files[1]['locations'] = set(['blenheim'])
-        jobFactory = splitter(package = "WMCore.DataStructs",
-                              subscription = testSubscription)
-        jobGroups = jobFactory(events_per_job = 50,
-                               halt_job_on_file_boundaries = True,
-                               performance = self.performanceParams,
-                               applyLumiCorrection = True)
+        jobFactory = splitter(package="WMCore.DataStructs",
+                              subscription=testSubscription)
+        jobGroups = jobFactory(events_per_job=50,
+                               halt_job_on_file_boundaries=True,
+                               performance=self.performanceParams,
+                               applyLumiCorrection=True)
 
         self.assertEqual(len(jobGroups), 1)
         jobs = jobGroups[0].jobs
         self.assertEqual(len(jobs), 3)
-
 
 
 if __name__ == '__main__':

